src/pages/index.tsx

```diff
@@ -69,8 +71,33 @@ function IndexPage() {
     },
     validate,
     onSubmit: (values) => {
-      console.log(values);
-      setFormSubmitted(true);
+      fetch("http://139.59.154.199:49160/api/v1/leads", {
+        method: "POST",
+        headers: {
+          Authorization: "secret_token",
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          name: values.username,
+          email: values.email,
+          consentsAccepted: values.acceptance,
+        }),
+      })
+        .then((response) => {
+          if (response.ok) {
+            setFormSubmitted(true);
+          } else if (response.status >= 500 && response.status < 600) {
+            setError5xx(true);
+          } else {
+            response.json().then((data) => {
+              setErrorMessage(data.message);
```

Comment:
**issue:** But you should not `setErrorMessage` on correctly fetched data. Btw, you probably won't have `data.message` structure here. It depends on your backend response. Check your request in `network` tab, to see that is inside. IMO you don't need to set anything here